test/main.js

```diff
@@ -666,6 +666,150 @@ describe('glob-stream', function() {
+      expect(pathObjs.length).toEqual(1);
+      expect(pathObjs[0]).toEqual(expected);
+    }
+
+    pipe([
+      globStream('\\!case', { cwd: cwd }),
+      concat(assert),
+    ], done);
+  });
+
+  it('should handle pattern starting with exclam and paren', function(done) {
+    var cwd = dir + '/fixtures/edge';
+    var expected = {
+      cwd: cwd,
+      base: cwd,
+      path: cwd + '/!(case)',
+    };
+
+    function assert(pathObjs) {
+      expect(pathObjs.length).toEqual(1);
+      expect(pathObjs[0]).toEqual(expected);
+    }
+
+    pipe([
+      globStream('\\!(case)', { cwd: cwd }),
+      concat(assert),
+    ], done);
+  });
+
+  it('should handle pattern that is a negative extglob', function(done) {
+    var cwd = dir + '/fixtures/edge';
+
+    function assert(pathObjs) {
+      expect(pathObjs.length).toEqual(3);
+      pathObjs = pathObjs.filter(function(pathObj) {
+        return pathObj.path.substr(cwd.length) === '/case';
+      });
+      expect(pathObjs.length).toEqual(0);
+    }
+
+    pipe([
+      globStream('!(case)', { cwd: cwd }),
+      concat(assert),
+    ], done);
+  });
+
+  it('should ignore pattern starting with paren', function(done) {
+    var cwd = dir + '/fixtures/edge';
+
+    function assert(pathObjs) {
+      expect(pathObjs.length).toEqual(3);
+      pathObjs = pathObjs.filter(function(pathObj) {
+        return pathObj.path.substr(cwd.length) === '/(case)';
+      });
+      expect(pathObjs.length).toEqual(0);
+    }
+
+    pipe([
+      globStream(['*', '!\\(case)'], { cwd: cwd }),
+      concat(assert),
+    ], done);
+  });
+
+  it('should ignore pattern starting with exclam', function(done) {
+    var cwd = dir + '/fixtures/edge';
+
+    function assert(pathObjs) {
+      expect(pathObjs.length).toEqual(3);
+      pathObjs = pathObjs.filter(function(pathObj) {
+        return pathObj.path.substr(cwd.length) === '/!case';
+      });
+      expect(pathObjs.length).toEqual(0);
+    }
+
+    pipe([
+      globStream(['*', '!\\!case'], { cwd: cwd }),
+      concat(assert),
+    ], done);
+  });
+
+  it('should ignore pattern starting with exclam and paren', function(done) {
+    var cwd = dir + '/fixtures/edge';
+
+    function assert(pathObjs) {
+      expect(pathObjs.length).toEqual(3);
+      pathObjs = pathObjs.filter(function(pathObj) {
+        return pathObj.path.substr(cwd.length) === '/!(case)';
+      });
+      expect(pathObjs.length).toEqual(0);
+    }
+
+    pipe([
+      globStream(['*', '!\\!(case)'], { cwd: cwd }),
+      concat(assert),
+    ], done);
+  });
+
+  it('should ignore pattern that is a negative extglob', function(done) {
+    var cwd = dir + '/fixtures/edge';
+    var expected = {
+      cwd: cwd,
+      base: cwd,
+      path: cwd + '/case',
+    };
+
+    function assert(pathObjs) {
+      expect(pathObjs.length).toEqual(1);
+      expect(pathObjs[0]).toEqual(expected);
+    }
+
+    pipe([
+      globStream(['*', '!!(case)'], { cwd: cwd }),
```

Comment:
I just finished running the tests on Windows 8, Windows 10 and mac with node.js 6 and the current version, and everything is passing 100%. I also logged out the paths created for the failing unit tests, and everything looks correct to me. 

I'm waiting for nvm to finish installing on my windows box so I can test other versions. I'll update here when I get those results. 

![windows-8](https://cloud.githubusercontent.com/assets/383994/22756654/d22b4bfc-ee15-11e6-8334-1ac665f05538.png)